helix-term/src/commands.rs

```diff
@@ -4014,16 +4013,50 @@ pub mod insert {
         doc.apply(&transaction, view.id);
     }
 
-    // TODO: handle indent-aware delete
     pub fn delete_char_backward(cx: &mut Context) {
         let count = cx.count();
         let (view, doc) = current!(cx.editor);
         let text = doc.text().slice(..);
         let transaction =
             Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
                 let pos = range.cursor(text);
+                let line_start_pos = text.line_to_char(range.cursor_line(text));
+
+                // considier to delete by indent level if all characters before `pos` are indent units.
+                let indent_unit = doc.indent_unit();
+                let line_to_start = text.slice(line_start_pos..pos).as_str().unwrap();
```

Comment:
This unwrap isn't safe. We also don't want to allocate if the string is in a single chunk:

```suggestion
                let line_to_start = Cow::from(text.slice(line_start_pos..pos));
```